bin/.common.sh

```diff
@@ -0,0 +1,31 @@
+# shellcheck shell=bash
+
+log() {
+  _fg=''
+  [ "${1}" = "red" ] && _fg='\033[0;31m'
+  [ "${1}" = "blue" ] && _fg='\033[0;34m'
+  [ "${1}" = "green" ] && _fg='\033[0;32m'
+  shift
+  # shellcheck disable=SC2145
+  echo -e "${_fg}>>> ${@}\033[0m"
+}
+
+_checkenv(){
```

Comment:
I think you could avoid needing _checkenv if you `set -euo pipefail` at the top of your script.

More info here: http://redsymbol.net/articles/unofficial-bash-strict-mode/

Reply:
Absolutely, the reason `_checkenv` was used in the first place is to get a more useful message, saying "you need *all* these variables set".

Happy to change, it will reduce complexity.